mathesar_ui/src/systems/table-view/header/header-cell/utils.ts

```diff
@@ -0,0 +1,33 @@
+import type { CellDataType } from '@mathesar/components/cell-fabric/data-types/typeDefinitions';
+import { SortDirection } from '@mathesar/stores/table-data';
+
+export function getSortingLabelForColumn(
+  dataType: CellDataType,
+  hasFkConstraint: boolean,
+): Record<SortDirection, string> {
+  if (['string', 'uri'].includes(dataType) || hasFkConstraint) {
```

Comment:
I like that you've incorporated the presence of the FK constraint into the logic used to label the sorting menu entries, but I don't really like the outcome of the logic you've used.

Take this example:

1. Here are my Publishers:

    ![image](https://user-images.githubusercontent.com/42411/212109861-58d2e8cb-f8d8-4289-b223-7b7ca7eb984c.png)

1. When I sort Publications by Publisher via "A-Z", I end up with "Cherry" first and "Apple" last.

    ![image](https://user-images.githubusercontent.com/42411/212109151-7afaa1cf-3af1-4ddc-a113-927a165c64e9.png)

    That's because it's sorting by the numerical value, not the record summary.
    
Unfortunately, because of our early decision to delegate record summary rendering almost entirely to the front end, I would venture to say that reliably sorting by record summary will be nearly impossible for us. I suspect that we'd need considerable backend work even to handle simple cases well.

This presents a UX challenge for us: **How do we communicate the behavior of FK sorting to the user?**

Perhaps the following menu labels would suffice?

- > Sort by ascending primary key

- > Sort by descending primary key

This is tricky though.

@kgodey @ghislaineguerin do you have any ideas here?

Reply:
I would use the labels:

> Sort by ascending ID
> Sort by descending ID

I can think of more complicated solutions, but doesn't seem worth the time investment right now.

Reply:
8b8e515ee

Reply:
@rajatvijay Can you remove the text I have crossed out in this image?

![image](https://user-images.githubusercontent.com/42411/212983449-b7b0b82f-e98c-436a-a0ca-5ef1c0d91697.png)

Reply:
58514ecc4